Client/Frontend/Browser/LightweightThemeManager.swift

```diff
@@ -0,0 +1,121 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import UIKit
+import SnapKit
+import Shared
+
+private var imgViews = WeakList<UIImageView>()
+private var instance: LightweightThemeManager? = nil
+
+@objc
+public class LightweightThemeManager {
+    private var profile: Profile
+
+    init(profile: Profile) {
+        self.profile = profile
+
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "SELThemeImageChanged:", name: "ThemeImageChanged", object: nil)
+    }
+
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: "ThemeImageChanged", object: nil)
+    }
+
+    private func getThemeLightness(img: UIImage?) {
+        if let img = img {
+            // Get a list of colors near the top of the image
+            let width = Int(img.size.width)
```

Comment:
Nit: It would be cool if we could have this sample color code as an extension to UIImage